src/utils/merge-config.ts

```diff
@@ -1,16 +1,23 @@
-import { Configuration } from '../types'
+import { mergeObjects } from './helper'
+import { FullConfiguration } from '../types'
 
-export default function mergeConfig(
-  a: Partial<Configuration> | null,
-  b: Partial<Configuration> | null
+export function mergeConfigs(
+  a: Partial<FullConfiguration>,
+  b?: Partial<FullConfiguration>
 ) {
-  const v: Partial<Configuration> = { ...a, ...b }
+  // Need to create a new object to avoid mutating the original here.
+  const v: Partial<FullConfiguration> = mergeObjects(a, b)
 
-  const m1 = a && a.middlewares
-  const m2 = b && b.middlewares
+  if (!b) return v
+
+  const { middlewares: m1, fallbackValues: f1 } = a
+  const { middlewares: m2, fallbackValues: f2 } = b
   if (m1 && m2) {
     v.middlewares = m1.concat(m2)
   }
+  if (f1 && f2) {
+    v.fallbackValues = mergeObjects(f1, f2)
```

Comment:
consider use `fallback` instead of `fallbackValues`? it's aligned with React naming. Will be easy for devs to understand

Reply:
Yeah I'm struggling with the naming here. `fallback` is great but it's singular, and `fallbacks`/`fallbackMap` feel weird...

Reply:
imo `fallback` doesn't have to imply singular, especially if the examples showed it was an object with multiple keys

Reply:
Which name would you prefer as an API option (since you're a native speaker)?

My original design was:
- `fallbackValues` for a object with multiple keys
- `fallbackValue` for a single value (same as initialData)

If we choose `fallback`, we can have both of these^.